procurement_purchase_no_grouping/models/stock_rule.py

```diff
@@ -15,16 +16,19 @@ class StockRule(models.Model):
     def _run_buy(self, product_id, product_qty, product_uom, location_id,
                  name, origin, values):
         grouping = product_id.categ_id.procured_purchase_grouping
-        self_wc = self.with_context(grouping=grouping)
+        self_wc = self.with_context(grouping=grouping, skip_domain=False)
+        if grouping == 'product_category':
+            self_wc = self.with_context(grouping=product_id.categ_id, skip_domain=True)
         return super(StockRule, self_wc)._run_buy(
             product_id, product_qty, product_uom, location_id, name,
             origin, values)
 
     def _make_po_get_domain(self, values, partner):
         domain = super()._make_po_get_domain(values, partner)
-        if self.env.context.get('grouping', 'standard') == 'order':
-            if values.get("move_dest_ids"):
-                domain += (("id", "=", -values["move_dest_ids"][:1].id),)
-            # The minimum is imposed by PG int4 limit
-            domain += (("id", "=", random.randint(-2147483648, 0)),)
+        if not self.env.context.get('skip_domain'):
+            if self.env.context.get('grouping', 'standard') == 'order':
+                if values.get("move_dest_ids"):
+                    domain += (("id", "=", -values["move_dest_ids"][:1].id),)
+                # The minimum is imposed by PG int4 limit
+                domain += (("id", "=", random.randint(-2147483648, 0)),)
```

Comment:
What kind of magic is being done here? negative ids? O_o
```suggestion
        if not self.env.context.get('skip_domain') and self.env.context.get('grouping', 'standard') == 'order':
            if values.get("move_dest_ids"):
                domain += (("id", "=", -values["move_dest_ids"][:1].id),)
            # The minimum is imposed by PG int4 limit
            domain += (("id", "=", random.randint(-2147483648, 0)),)
```

Reply:
The trick is used for having no results at all from the search.